File: CS/XtraTabWithButtons/ViewInfo/Other/SkinViewInfoRegistratorDescendant.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraTab.Registrator;
using DevExpress.XtraTab.ViewInfo;
using DevExpress.XtraTab;
using DevExpress.XtraTab.Drawing;

namespace XtraTabWithButtons
{
    public class SkinViewInfoRegistratorDescendant : SkinViewInfoRegistrator
    {
        public const string MyViewName = "MySkin";
        public override string ViewName
        {
            get
            {
                return MyViewName;
            }
        }

        public override BaseTabHandler CreateHandler(IXtraTab tabControl)
        {
            return new BaseTabHandlerDescendant(tabControl);
        }

        public override BaseTabHeaderViewInfo CreateHeaderViewInfo(BaseTabControlViewInfo viewInfo)
        {
            return new SkinTabHeaderViewInfoDescendant(viewInfo);
        }

        public override BaseTabPainter CreatePainter(IXtraTab tabControl)
        {
            return new SkinTabPainterDescendant(tabControl);
        }
    }
}

File: VB/XtraTabWithButtons/ViewInfo/Other/SkinViewInfoRegistratorDescendant.vb
Imports DevExpress.XtraTab.Registrator
Imports DevExpress.XtraTab.ViewInfo
Imports DevExpress.XtraTab
Imports DevExpress.XtraTab.Drawing

Namespace XtraTabWithButtons

    Public Class SkinViewInfoRegistratorDescendant
        Inherits SkinViewInfoRegistrator

        Public Const MyViewName As String = "MySkin"

        Public Overrides ReadOnly Property ViewName As String
            Get
                Return MyViewName
            End Get
        End Property

        Public Overrides Function CreateHandler(ByVal tabControl As IXtraTab) As BaseTabHandler
            Return New BaseTabHandlerDescendant(tabControl)
        End Function

        Public Overrides Function CreateHeaderViewInfo(ByVal viewInfo As BaseTabControlViewInfo) As BaseTabHeaderViewInfo
            Return New SkinTabHeaderViewInfoDescendant(viewInfo)
        End Function

        Public Overrides Function CreatePainter(ByVal tabControl As IXtraTab) As BaseTabPainter
            Return New SkinTabPainterDescendant(tabControl)
        End Function
    End Class
End Namespace
